src/trace/listener.ts

```diff
@@ -107,6 +118,11 @@ export class TraceListener {
       ) {
         options.tags["_dd.parent_source"] = this.contextService.traceSource;
       }
+      if (this.triggerTags) {
+        options.tags = { ...options.tags, ...this.triggerTags };
+        // # Add trigger tags under the dd subsegment's root_span_metadata field
+        addXrayMetadata(xrayBaggageSubsegmentKey, this.triggerTags);
```

Comment:
Is this the right place to call this function? I feel the surrounding code only handles the DD native tracing mode. Perhaps should be in `onStartInvocation`?